### PR TITLE
Update speaker.update with speech_state and permissions, update tests.

### PR DESCRIPTION
**Pull request by reiterl**

### CODE REVIEW

**Review by jsangmeister — CHANGES_REQUESTED** [review]
Code looks good, but please add more tests:
- a working and a failing one for each of the entries in the table in https://github.com/OpenSlides/OpenSlides/wiki/speaker.update (X -> pro/contra working and X -> contribution working are already done, rest is missing)
- more permission tests:
  - request user has CAN_SEE, but is not speaker user -> fails
  - request user has CAN_SEE and is speaker user -> works
  - request user is speaker user, but does not have can_see -> fails

**Review by jsangmeister — CHANGES_REQUESTED** [review]
Also, please split the `speaker/create_update_delete.py` into seperate files, they do not belong together

File: openslides_backend/action/actions/speaker/create_update_delete.py [hunk 1]
```diff
@@ -199,6 +199,65 @@ class SpeakerUpdate(UpdateAction):
...
+            self.user_id,
+            Permissions.ListOfSpeakers.CAN_SEE,
+            speaker["meeting_id"],
+        ):
+            return
+        super().check_permissions(instance)
+
 
 @register_action("speaker.delete")
 class SpeakerDeleteAction(DeleteAction):
```
Comment (FinnStutzenstein) [inline openslides_backend/action/actions/speaker/create_update_delete.py:263]: Can you separate the classes into their own files?

File: openslides_backend/action/actions/speaker/create_update_delete.py [hunk 2]
```diff
@@ -199,6 +199,65 @@ class SpeakerUpdate(UpdateAction):
...
+        allowed_self_contribution = has_can_manage or meeting.get(
+            "list_of_speakers_can_set_contribution_self"
+        )
+        allowed_pro_contra = meeting.get("list_of_speakers_enable_pro_contra_speech")
+        msg = "Not allowed to set speech_state"
+        if speaker.get("speech_state") == instance.get("speech_state"):
+            pass
+        elif instance.get("speech_state") == "contribution":
+            if not allowed_self_contribution:
+                raise ActionException(msg)
```
Comment (FinnStutzenstein) [inline openslides_backend/action/actions/speaker/create_update_delete.py:229]: Maybe a fitting message describing the situation would be better. E.g. here: "Self contributing is not enabled."

**Review by FinnStutzenstein — CHANGES_REQUESTED** [review]
Code looks good. Maybe add a few more tests.

**Comment by reiterl** [comment]
Split create_update_delete in single action files.

File: openslides_backend/action/actions/speaker/create_update_delete.py (hunk 2 above)
Comment (reiterl) [reply openslides_backend/action/actions/speaker/create_update_delete.py]: Done.

**Review by FinnStutzenstein — APPROVED** [review]

**Review by jsangmeister — APPROVED** [review]

**Pull request merged**